src/mocks/mockISelectionManager.ts

```diff
@@ -55,25 +55,46 @@ export class MockISelectionManager implements ISelectionManager {
     }
 
     public select(selectionId: ISelectionId | ISelectionId[], multiSelect?: boolean): IPromise<ISelectionId[]> {
-        const selectionIds: ISelectionId[] = [].concat(selectionId);
+        const selectionIds: ISelectionId[] = Array.isArray(selectionId) ? selectionId : [selectionId];
 
-        // if no multiselect reset current selection and save new passed selections;
-        if (!multiSelect) {
-            this.selectionIds = selectionIds;
+        if (selectionIds.length < 1) {
+            return new Promise((resolve, reject) => {
+                resolve(this.selectionIds);
+            }) as any;
+        }
+
```

Comment:
I see a lot of `If else` blocks and all of them have the same `this.selectionIds = selectionIds`

Reply:
If this is right, you can make it much easier:
```js
if(multiSelect){
...all your ifs here
} else {
   this.selectionIds = selectionIds
}

```

Reply:
It's hard as the logic for single/multiple selection is different.